aws/resource_aws_lb_target_group.go

```diff
@@ -273,21 +273,25 @@ func resourceAwsLbTargetGroupUpdate(d *schema.ResourceData, meta interface{}) er
 			healthCheck := healthChecks[0].(map[string]interface{})
 
 			params = &elbv2.ModifyTargetGroupInput{
-				TargetGroupArn:        aws.String(d.Id()),
-				HealthCheckPort:       aws.String(healthCheck["port"].(string)),
-				HealthCheckProtocol:   aws.String(healthCheck["protocol"].(string)),
-				HealthyThresholdCount: aws.Int64(int64(healthCheck["healthy_threshold"].(int))),
+				TargetGroupArn:          aws.String(d.Id()),
+				HealthCheckPort:         aws.String(healthCheck["port"].(string)),
+				HealthCheckProtocol:     aws.String(healthCheck["protocol"].(string)),
+				HealthyThresholdCount:   aws.Int64(int64(healthCheck["healthy_threshold"].(int))),
+				UnhealthyThresholdCount: aws.Int64(int64(healthCheck["unhealthy_threshold"].(int))),
 			}
 
 			healthCheckProtocol := strings.ToLower(healthCheck["protocol"].(string))
 
+			if !d.IsNewResource() && d.HasChange("health_check.0.interval") && healthCheckProtocol == "tcp" {
```

Comment:
From our chat just now I learned that `ResourceDiff.ForceNew` doesn't accept dot-separated paths like this and that's why you are doing this check in here.

The downside of doing it in here is that it won't fail until apply time. A possible compromise is to do this _check_ inside `CustomizeDiff` and have it return an error (rather than calling `ForceNew`) if it detects a change like this. That way it will fail during plan time rather than apply time.

Not ideal of course, but I think better than generating a plan successfully and then bailing out partway through apply.

Reply:
Ha, that's a good idea actually 🤔  I'll change it.